dvc/tree/base.py

```diff
@@ -220,6 +220,27 @@ def walk_files(self, path_info, **kwargs):
     def is_empty(self, path_info):
         return False
 
+    def _getsize(self, path_info):
+        return None
+
+    def getsize(self, path_info):
+        if self.isfile(path_info):
+            return self._getsize(path_info)
+
+        try:
+            file_iter = self.walk_files(path_info)
```

Comment:
I guess this becomes more of `du` than `getsize` at this point. In transfer you'll likely need to walk and transfer at the same time, and walking files here once again will be wasteful in terms of API calls.

Reply:
If you only really need getsize for files right now, I would just stick to that. `du` will be useful later, but there seems to be no need to implement it right now.

Reply:
Sure! Since I will always pass a file in the end, it wouldn't matter much for me. Will only make this available for files